code/game/gamemodes/dynamic/dynamic_maths.dm

```diff
@@ -0,0 +1,48 @@
+// Exponential distribution.
+// Used for midround/latejoin injection delays.
+/proc/exp_distribution(var/desired_mean)
+	if (desired_mean <= 0)
+		desired_mean = 1
+	var/lambda = 1/desired_mean
+	var/x = rand()
+	while (x == 1)
+		x = rand()
```

Comment:
Drop the while and do something like
`rand(0, 999) * 0.00.1` or however many significant digits you want